Better "select" & "select_multiple" Scrolling.

therealOri opened the pull request:
# Updated:
- Allows the user to press down when at the bottom of a list to send the cursor/index back to the top.
- Ditto ^^ but for when the user is at the top of a list and presses up. 

<br />

# Left alone:
- I didn't touch the "confirm" function as it is just 2 options and is basically a True or False check and I like it how it is.
- I didn't touch the "prompt" function as it is unrelated.

<br />
<br />

# Why?:
When you start to have pretty BIG lists, say 20 options for example...it will take some time to get all the way back to the top of the list of options. Same for when you want to get to the bottom of the list. I would have to tap the down arrow key like 19 times to get to the bottom. I can now just press up 1 time and BAM, bottom of the list. (Same for being at the bottom and getting back to the top) 
__ __

<br />
<br />


# Tested with:
> Successfully. 
- Python Version: v3.10
- OS: Manjaro Linux x86_64
- Kernel: 5.15.85-1-MANJARO 
__ __

therealOri commented:
I am unsure as to why `poetry run poe test` would result in 3 failures. Maybe it's an issue with the tests themself expecting a different result?

![2023-01-04_10-01](https://user-images.githubusercontent.com/45724082/210609736-f22c5049-bd57-432f-8085-035463b9d089.png)
![2023-01-04_10-01_1](https://user-images.githubusercontent.com/45724082/210609766-421640de-bd9e-4540-b267-811ded8bc4eb.png)
![2023-01-04_10-02](https://user-images.githubusercontent.com/45724082/210609819-50d0230f-efae-4436-af22-b236a6102ee3.png)
![2023-01-04_10-02_1](https://user-images.githubusercontent.com/45724082/210609831-6ac3c2c8-afe8-4fb1-bbe3-d6502e826703.png)
__ __

<br />

I took a look at the steps that are automated and when I do them myself, I get exactly what I expected and want. Maybe I don't understand the tests. idk.

petereon commented:
Hey @therealOri , thanks for the contribution. You are right! It is tests that are not adjusted to new behavior. Tests are also in dire need of rework as testing of the console output makes them heavily implementation dependent and most are pretty much intertwined at the moment which makes adding new functionality and changing existing one a bit of a a hassle.

Furthermore, I believe there are instances of tests where I was testing exactly that it does not "overflow". This might be causing issues now that it does. 

I am currently pretty pressed for time but I can have a proper look on Friday I think.

therealOri commented:
Huh..Even though I haven't changed a single thing with the code since last night when I made the pull request, it doesn't seem to be working now..I'll look more into this later.

petereon reviewed (changes requested):
Awaiting merge of [#1](https://github.com/therealOri/beaupy/pull/1) or comparable changes.

therealOri commented:
The patch I have applied to therealOri/beaupy that you have requested seems to have reverted `_beaupy.py` back to it's original state before my changes. 

I have updated it again in my repository. So just make sure that when you decide to merge code, that what's being merged is what is supposed to be merged.

petereon commented:
> The patch I have applied to therealOri/beaupy that you have requested seems to have reverted `_beaupy.py` back to it's original state before my changes.
> 
> I have updated it again in my repository. So just make sure that when you decide to merge code, that what's being merged is what is supposed to be merged.

What do you mean reverted? Granted, the `if` blocks are gone, because I have refactored the changes in the `_beaupy.py` to use modulo instead of conditionals.

therealOri commented:
> > The patch I have applied to therealOri/beaupy that you have requested seems to have reverted `_beaupy.py` back to it's original state before my changes.
> > I have updated it again in my repository. So just make sure that when you decide to merge code, that what's being merged is what is supposed to be merged.
> 
> What do you mean reverted? Granted, the `if` blocks are gone, because I have refactored the changes in the `_beaupy.py` to use modulo instead of conditionals.

Ohh, Okay.

petereon commented:
Sorry if I wasn't awfully transparent about that. But for me it seems to work, whenever it overflows the modulo bumps it to the other end of acceptable range of indexes.

therealOri commented:
> Sorry if I wasn't awfully transparent about that. But for me it seems to work, whenever it overflows the modulo bumps it to the other end of acceptable range of indexes.

It does work for me as well. And it's alright, I wasn't reading closely enough xD.

petereon commented:
> > Sorry if I wasn't awfully transparent about that. But for me it seems to work, whenever it overflows the modulo bumps it to the other end of acceptable range of indexes.
> 
> It does work for me as well. And it's alright, I wasn't reading closely enough xD.

Perfect, I'll run some manual testing on Windows, hopefully get the lint action to pass and then I'll merge and release! Thanks for contribution once again!
